### PR TITLE
community[minor]: Add graph store extractors

### PR DESCRIPTION
**Pull request by cbornet**
This adds an extractor interface and an implementation for HTML pages.
Extractors are used to create GraphVectorStore Links on loaded content.

**Twitter handle:** cbornet_

### CODE REVIEW

File: libs/community/langchain_community/graph_vectorstores/extractors/html_link_extractor.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        parse_url = _parse_url(link, page_url=url, drop_fragments=drop_fragments)
+        # Remove self links and entries for any 'a' tag that failed to parse
+        # (didn't have href, or invalid domain, etc.)
+        if parse_url and parse_url != url:
+            links.add(parse_url)
+
+    return links
+
+
+@dataclass
```
Comment (eyurtsev) [inline libs/community/langchain_community/graph_vectorstores/extractors/html_link_extractor.py:58]: If possible
```suggestion
@dataclass(frozen=True)
```

**Comment by eyurtsev** [comment]
@cbornet if you're confident with the abstractions we can also commit it to `langchain` which is meant for provider agnostic implementations.

**Review by eyurtsev — APPROVED** [review]
Looks good to me overall -- going to give it another look likely today and then merge

File: libs/community/langchain_community/graph_vectorstores/extractors/html_link_extractor.py (hunk 1 above)
Comment (bjchambers) [reply libs/community/langchain_community/graph_vectorstores/extractors/html_link_extractor.py:58]: Should be possible.

**Pull request merged**